src/librustc_middle/mir/mod.rs

```diff
@@ -2258,6 +2263,10 @@ impl<'tcx> Debug for Rvalue<'tcx> {
             UnaryOp(ref op, ref a) => write!(fmt, "{:?}({:?})", op, a),
             Discriminant(ref place) => write!(fmt, "discriminant({:?})", place),
             NullaryOp(ref op, ref t) => write!(fmt, "{:?}({:?})", op, t),
+            Tls(did) => ty::tls::with(|tcx| {
```

Comment:
I found this pretty surprising -- why not something like `Tls(&mut X)`? Otherwise, it looks exactly like a regular borrow expression, no?

Reply:
It does, I was considering adding *some* annotation that this is a tls access, but I wasn't sure how to do that in a nice syntax.

Reply:
Since our surface syntax doesn't distinguish this (unfortunately), if we want to stick to valid surface syntax, I think we best we can do is add a comment:
```rust
&mut NAME /* for current thread */
&mut NAME /* thread-local */
```
```

Reply:
Or maybe `&tls mut NAME`?

Reply:
I like that, similar to `&raw` references.

Oh. I just now saw that you meant to insert a comment there. How do you like `&tls mut NAME` and `&tls NAME`?

Reply:
Well it's not actual syntax... so I think `/*tls*/ is better.

Reply:
I was originally thinking about `&tls mut NAME`, but then saw that @RalfJung said `if we want to stick to valid surface syntax`, so I changed it to a comment.